packages/integrations/mdx/src/recma-inject-import-meta-env.ts

```diff
@@ -11,7 +11,7 @@ export function recmaInjectImportMetaEnv({
 			if (node.type === 'MemberExpression') {
 				// attempt to get "import.meta.env" variable name
 				const envVarName = getImportMetaEnvVariableName(node);
-				if (typeof envVarName === 'string') {
+				if (typeof envVarName === 'string' && importMetaEnv[envVarName] != null) {
```

Comment:
This check was the main reason for the error in https://github.com/withastro/astro/issues/9012, since if it's undefined, we accidentally generate a broken AST.

After fixing this, it relies on other fixes in this PR so that it works correctly in runtime.